### ISSUE TITLE
Unable to fetch GSF ID

### ISSUE THREAD

**Issue by JollyPanda823** [GrapheneOS/os-issue-tracker#4574]
**Device:** Pixel 8
**GrapheneOS version:** 2024121200
**Google Play Service version:** 24.47.36
**GmsCompatConfig version:** 152

I am unable to extract my Google Services Framework (GSF) ID in order to register my device for Play Protect Certification exemption ([google.com/android/uncertified](google.com/android/uncertified)). I tried using [Castro](https://play.google.com/store/apps/details?id=com.itemstudio.castro) and [kDI Device Info](https://f-droid.org/packages/com.oF2pks.kalturadeviceinfos/). In Castro, the ID doesn't show up as a category and in kDI Device Info it is said to be unknown. I am logged into my Google Account. I know I could see the GSF ID in previous versions of GrapheneOS as I already did the Play Protect Certification exemption registration on a previous installation.

**Comment by MichaelAnders** [comment]
Pixel 8 pro

Using grapheneos since a year without issues - cool!

I need an app now (electroverse) for charging my new EV.

Unfortunately, I also need the GSF ID so I can give it Google and get the app installed.

Indeed, no app out there seems to work as JollyPanda823 stated.

That might be a security feature of Grapheneos?

Anyways, what I'd like to see is an option in the Android menue to display it so I can copy paste it. As you have the code, adding a hook or something to fetch this should be possible somehow I hope?

A comment from the devs would be very much appreciated (open/checking/later/impossible), and thanks a million for the work you guys put into this really great OS!

**Comment by mpiraz** [comment]
I hope electroverse app https://play.google.com/store/apps/details?id=energy.octopus.electricjuice.android  (for managing EV car) will work asap with grapheneos. I need to use it but it doesn't work as I verified yesterday during grapheneos configuration process in my new Pixel 9a

**Comment by mateibarbu19** [comment]
@JollyPanda823 did you manage to find an answer to your question?
I am also interested in the answer. Thanks.

**Comment by adrianer** [comment]
It's also not about the install source checking. It's about checking if the device is "Play Protect" certified. Not sure, though, if GrapheneOS isn't already faking that check - at least it isn't faking it for the Play Store itself, as it reports the device as uncertified.

**Comment by adrianer** [comment]
@thestinger : I don't think this is related, because on my other phone with LineageOS, which is not even passing BASIC Play Integrity level, after setting up the exemption on that Google site I get that my device is Play Protect certified in the About section of the Google Store.

But yes: this certification does not do anything for apps that check the Play Integrity API and solely based on that refuse to work.

**Comment by thestinger** [comment]
@adrianer What reason is there to add an exemption? It doesn't appear to help anything since apps using the Play Integrity API for their services or Play Store listing. There's little reason for people to use this exemption system if it doesn't solve anything.

I think GSF ID is inaccessible to user installed apps on modern Android. That has nothing to do with GrapheneOS and is why the issue is invalid, since it's not specific to GrapheneOS and not a valid way to retrieve the GSF ID. I don't think there's a way to retrieve the GSF ID on a production build due to lack of adb root access unless it's shown somewhere in the Google Play UI.
